packages/webapp/src/containers/Map/LocationSelectionModal/index.jsx

```diff
@@ -5,16 +5,24 @@ import React from 'react';
 import { useSelector } from 'react-redux';
 import { canShowSelectionSelector, mapLocationsSelector } from '../../mapSlice';
 import PurePreviewPopup from '../../../components/Map/PreviewPopup';
+import { SENSOR } from '../../SensorReadings/constants';
 
-export default function LocationSelectionModal({ history, selectingOnly }) {
+export default function LocationSelectionModal({ history, sensorReadings, selectingOnly }) {
   const { dismissSelectionModal } = useSelectionHandler();
   const showSelection = useSelector(canShowSelectionSelector);
   const locations = useSelector(mapLocationsSelector);
-  if (showSelection && locations.length === 1 && locations[0].type === 'sensor') {
+  const locationSensorReadings = Object.values(sensorReadings).filter(
```

Comment:
You might want to wrap this in a `useMemo`. Also why are we only considering the first location in the list?

Reply:
The array only has one location unless it's a cluster, in which case, we use all the locations. So for previewing one sensor, the locations will have one item (it would make sense more if it wasn't an array but we need it to be an array for the clusters).